Build/Build.cs

```diff
@@ -258,11 +272,36 @@ class Build : NukeBuild
         });
 
     Target SpellCheck => _ => _
+        .OnlyWhenDynamic(() => DocumentationChangesDetected())
         .Executes(() =>
         {
             Node($"{YarnCli} install --silent", workingDirectory: RootDirectory);
             Node($"{YarnCli} --silent run cspell --no-summary", workingDirectory: RootDirectory,
                 customLogger: (_, msg) => Error(msg));
         });
+
+    bool DocumentationChangesDetected()
+    {
+        return Changes.Any(x => x.StartsWith("docs"));
+    }
+
+    bool SourceChangesDetected()
+    {
+        return Changes.Any(x => !x.StartsWith("docs"));
+    }
+
+    IEnumerable<string> Changes
+    {
+        get
+        {
+            using var repo = new Repository(GitRepository.LocalDirectory);
+
+            return repo.Diff.Compare<TreeChanges>(repo.Branches["develop"].Tip.Tree,
```

Comment:
🔧 Not all PRs target `develop`, so you'll need to get the target branch as a parameter from the GHA build.

Reply:
I had something in mind.. will check this out soon